### PR TITLE
py-software-engineers

### PR DESCRIPTION
**Pull request by muriakula**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1 +1,57 @@
...
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills = [
+            "Python",
+            "SQL",
+            "Django",
+            "JavaScript",
+            "HTML",
+            "CSS",
+        ]
```
Comment (Bohdan-Salamakha) [inline app/main.py:52]: don't hardcode skills
u need to extend skills in every class

File: app/main.py [hunk 2]
```diff
@@ -1 +1,57 @@
...
+
+    def create_powerful_api(self) -> str:
+        print(f"{self.name} is creating an API...")
+        return "http://127.0.0.1:8000"
+
+
+class AndroidDeveloper(SoftwareEngineer):
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills = ["Java", "Android studio"]
```
Comment (Bohdan-Salamakha) [inline app/main.py:33]: don't hardcode skills
u need to extend skills in every class

File: app/main.py [hunk 3]
```diff
@@ -1 +1,57 @@
...
+
+    def create_awesome_web_page(self) -> str:
+        print(f"{self.name} is creating a webpage...")
+        return "<h1>Hello world</h1>"
+
+
+class BackendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills = ["Python", "SQL", "Django"]
```
Comment (Bohdan-Salamakha) [inline app/main.py:23]: don't hardcode skills
u need to extend skills in every class

File: app/main.py [hunk 4]
```diff
@@ -1 +1,57 @@
...
+        self.skills = []
+
+    def learn_skill(self, skill: str) -> None:
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills = ["JavaScript", "HTML", "CSS"]
```
Comment (Bohdan-Salamakha) [inline app/main.py:13]: don't hardcode skills
u need to extend skills in every class

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
v1

File: app/main.py [hunk 5]
```diff
@@ -1 +1,52 @@
...
+    def learn_skill(self, skill: str) -> None:
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.learn_skill("JavaScript")
+        self.learn_skill("HTML")
+        self.learn_skill("CSS")
```
Comment (ArsenPidhoretskyi) [inline app/main.py:15]: Please read checklist. And use += or .extend

Comment (ArsenPidhoretskyi) [reply app/main.py:15]: Fix in all places

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

**Review by ArsenPidhoretskyi — APPROVED** [review]
Well done!

**Pull request closed without merging**